test/functional/wallet_reorgsrestore.py

```diff
@@ -90,6 +91,54 @@ def test_coinbase_automatic_abandon_during_startup(self):
         # Verify the coinbase descendant was also marked as abandoned
         assert_equal(wallet0.gettransaction(descendant_tx_id)['details'][0]['abandoned'], True)
 
+    def test_reorg_handling_during_unclean_shutdown(self):
+        self.log.info("Test that wallet doesn't crash due to a duplicate block disconnection event after an unclean shutdown")
+        node = self.nodes[0]
+        # Receive coinbase reward on a new wallet
+        node.createwallet(wallet_name="reorg_crash", load_on_startup=True)
+        wallet = node.get_wallet_rpc("reorg_crash")
+        self.generatetoaddress(node, 1, wallet.getnewaddress(), sync_fun=self.no_op)
+
+        # Restart to ensure node and wallet are flushed
+        self.restart_node(0)
+        wallet = node.get_wallet_rpc("reorg_crash")
+        assert_greater_than(wallet.getwalletinfo()['immature_balance'], 0)
+
+        # Disconnect tip and sync wallet state
+        tip = wallet.getbestblockhash()
+        wallet.invalidateblock(tip)
+        wallet.syncwithvalidationinterfacequeue()
```

Comment:
I would have used node RPCs here because these don't seem to be wallet specific requests and moreover, there's just one loaded wallet and using wallet RPCs here made me think there was some specific reason for it, but fine. I don't want the ACKs to be invalidated for this.

Reply:
will do if I have to retouch it.